yt_dlp/YoutubeDL.py

```diff
@@ -2466,11 +2484,11 @@ def restore_last_token(self):
         parsed_selector = _parse_format_selection(iter(TokenIterator(tokens)))
         return _build_selector_function(parsed_selector)
 
-    def _calc_headers(self, info_dict):
-        res = merge_headers(self.params['http_headers'], info_dict.get('http_headers') or {})
-        if 'Youtubedl-No-Compression' in res:  # deprecated
-            res.pop('Youtubedl-No-Compression', None)
-            res['Accept-Encoding'] = 'identity'
+    def _calc_headers(self, info_dict, load_cookies=True):
+        if load_cookies:  # For --load-info-json
+            self._load_cookies(traverse_obj(info_dict.get('http_headers'), 'Cookie', casesense=False),
+                               autoscope=info_dict['url'])  # compat
+            self._load_cookies(info_dict.get('cookies'), autoscope=False)
```

Comment:
Moving cookie loading into `_calc_headers` was the simplest way to catch cookies from both the format field `http_headers` and the info_dict root `http_headers`, and then scope them to format URL, set them to cookiejar, & add them to the `cookies` field